resources/waterheater.rb

```diff
@@ -759,22 +759,59 @@ def self.apply_combi_system_EMS(model, runner, combi_sys_id, dhw_map)
   end
 
   def self.add_desuperheater(model, t_set, tank, desuperheater_clg_coil, wh_type)
-    # Create a schedule for desuperheater control (schedule value - desuperheater deadband = a little bit over tank stp would be good)
-    new_schedule = OpenStudio::Model::ScheduleConstant.new(model)
-    new_schedule.setName("#{tank.name} desuperheater setpoint schedule")
-    new_schedule.setValue(UnitConversions.convert(t_set, "F", "C") + deadband(wh_type) / 2.0 + 0.5)
-
-    # create a desuperheater object
-    desuperheater = OpenStudio::Model::CoilWaterHeatingDesuperheater.new(model, new_schedule)
-    desuperheater.setDeadBandTemperatureDifference(0.2)
-    desuperheater.setRatedHeatReclaimRecoveryEfficiency(0.25)
-    desuperheater.addToHeatRejectionTarget(tank)
-    desuperheater.setWaterPumpPower(0)
+    reclaimed_efficiency = 0.25 # default
+    eta_c = tank.heaterThermalEfficiency.get
+    tank_name = tank.name.to_s.gsub(' ', '_')
+    output_array = []
+
+    coil_clg_energy = OpenStudio::Model::EnergyManagementSystemSensor.new(model, "Cooling Coil Total Cooling Energy")
+    coil_clg_energy.setName("#{desuperheater_clg_coil.name} clg energy")
+    coil_clg_energy.setKeyName(desuperheater_clg_coil.name.to_s)
+
+    coil_elec_energy = OpenStudio::Model::EnergyManagementSystemSensor.new(model, "Cooling Coil Electric Energy")
+    coil_elec_energy.setName("#{desuperheater_clg_coil.name} elec energy")
+    coil_elec_energy.setKeyName(desuperheater_clg_coil.name.to_s)
+
+    wh_energy = OpenStudio::Model::EnergyManagementSystemSensor.new(model, "Water Heater Heating Energy")
+    wh_energy.setName("#{tank.name} wh energy")
+    wh_energy.setKeyName(tank.name.to_s)
+
+    # energy global variables
+    dsh_total = OpenStudio::Model::EnergyManagementSystemGlobalVariable.new(model, "#{tank_name}_dsh_total")
+    wh_e = OpenStudio::Model::EnergyManagementSystemGlobalVariable.new(model, "#{tank_name}_e")
+    wh_htg = OpenStudio::Model::EnergyManagementSystemGlobalVariable.new(model, "#{tank_name}_htg_load")
+    dsh_e = OpenStudio::Model::EnergyManagementSystemGlobalVariable.new(model, "#{tank_name}_dsh_htg_load")
```

Comment:
Isn't `dsh_total` the only variable that needs to be a GlbalVariable? It is cumulative while all the others are instantaneous.

Reply:
Yes since we only want dsh fuel energy to be reported, these three and their output loop are removed.